backend/src/main/java/gov/cdc/usds/simplereport/service/DataHubUploaderService.java

```diff
@@ -0,0 +1,137 @@
+package gov.cdc.usds.simplereport.service;
+
+import com.fasterxml.jackson.dataformat.csv.CsvGenerator;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+import gov.cdc.usds.simplereport.api.model.TestEventExport;
+import gov.cdc.usds.simplereport.db.model.TestEvent;
+import gov.cdc.usds.simplereport.db.model.auxiliary.RaceArrayConverter;
+import gov.cdc.usds.simplereport.db.repository.TestEventRepository;
+import net.minidev.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import javax.persistence.NoResultException;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.text.SimpleDateFormat;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@Transactional
+public class DataHubUploaderService {
+    @Value("${simple-report.data-hub.uploadurl}")
+    private String DATAHUB_UPLOAD_URL;
+    static final int MAX_ROWS_ALLOWED_PER_BATCH = 999;
+    static final String API_VERSION = "2";
```

Comment:
Constants should be explicitly public or private—also, what API is this the version of?

Reply:
Good point. I'll make it private and I'll move `MAX_ROWS_ALLOWED_PER_BATCH` to the config file. 

The string is the version of the csv format from the client's point of view. It's only used so the data-hub folks know when what we're sending changes. (e.g. they can ask us when we'll be including column _X_ and we can say, it will be in version "5"). It's also likely that dev and prod will get out of sync code version wise as dev moves faster than prod.

This version string is being stuck into the http header.